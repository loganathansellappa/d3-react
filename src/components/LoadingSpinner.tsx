import React from 'react';
import './LoadingSpinner.scss';
interface LoadingSpinnerProps {
    size?: number;
    color?: string;
}

const LoadingSpinner: React.FC<LoadingSpinnerProps> = ({ size = 40, color = '#0073e6' }) => {
    const spinnerStyle: React.CSSProperties = {
        border: `4px solid ${color}`,
        borderTop: `4px solid transparent`,
        borderRadius: '50%',
        width: size,
        height: size,
        animation: 'spin 2s linear infinite',
    };

    return <div style={spinnerStyle} className="loading-spinner"></div>;
};
export default LoadingSpinner;
