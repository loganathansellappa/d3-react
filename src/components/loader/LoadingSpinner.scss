$spinner-color: #0073e6;

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.loading-spinner {
  display: inline-block;
  width: 64px;
  height: 64px;
  border: 6px solid $spinner-color;
  border-top: 6px solid transparent;
  border-radius: 50%;
  animation: spin 2s linear infinite;
}
