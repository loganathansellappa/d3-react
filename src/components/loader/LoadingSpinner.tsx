import React from "react";
import "./LoadingSpinner.scss";
interface LoadingSpinnerProps {
  size?: number;
  color?: string;
}

const LoadingSpinner: React.FC<LoadingSpinnerProps> = ({
  size = 40,
  color = "#0073e6",
}) => {
  const spinnerStyle: React.CSSProperties = {
    border: `4px solid ${color}`,
    borderTop: `4px solid transparent`,
    borderRadius: "50%",
    width: size,
    height: size,
    animation: "spin 2s linear infinite",
  };

  return (
    <>
      <div className="spinner-container">
        <div style={spinnerStyle} className="loading-spinner"></div>
      </div>
    </>
  );
};
export default LoadingSpinner;
