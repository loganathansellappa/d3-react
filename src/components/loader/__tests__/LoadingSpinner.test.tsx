import { render } from "@testing-library/react";
import LoadingSpinner from "../LoadingSpinner";

describe("LoadingSpinner", () => {
  let element: { container: HTMLElement };
  beforeEach(() => {
    element = render(<LoadingSpinner />);
  });

  test("renders KeyValueTable component", () => {
    expect(element.container.firstChild).toMatchSnapshot();
  });
});
