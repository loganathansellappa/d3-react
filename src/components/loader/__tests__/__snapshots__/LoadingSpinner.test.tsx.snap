// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner renders KeyValueTable component 1`] = `
<div
  class="spinner-container"
>
  <div
    class="loading-spinner"
    style="border: 4px solid #0073e6; border-top: 4px solid transparent; border-radius: 50%; width: 40px; height: 40px; animation: spin 2s linear infinite;"
  />
</div>
`;
